snorkel/contrib/candidate_viewer.py

```diff
@@ -0,0 +1,163 @@
+            if candidates_results:
+                # printing
+                for i in range(len(candidates_results)):
+                    arg1 = candidates_results[i][1]
+                    arg2 = candidates_results[i][2]
+                    arg1_text = self._get_span_text(arg1)
+                    arg2_text = self._get_span_text(arg2)
+
+                    if not len(keywords) or (any(k in arg1_text for k in keywords) or any(k in arg2_text for k in keywords)):
+                        print("[" + doc_name + "] " + arg1_text + " <-----> " + arg2_text)
+                        # check the max print for the output
+                        if i >= max_output - 1:
+                            break
+        except Exception as e:
+            print("[log-CandidateViewer] error in printing candidates. [Details]: {0}".format(str(e)))
+
+    def get_doc_text(self, doc_name):
+        """
+        getting text string of a document along with list of sentences and their starting indexes
+        :param doc_name: name of the document
+        :return: doc_text, doc_sents, doc_sents_start_idx
+        """
+        doc_sents = []
+        doc_sents_start_idx = []
+
+        tmp = self._db_select("SELECT id FROM document WHERE name = \"{0}\"".format(doc_name))
+        if tmp:
+            doc_id = tmp[0][0]
+            doc_sent_query = "SELECT stable_id FROM context WHERE type = \"sentence\" and stable_id like \"{0}::%\"".format(doc_name)
+            doc_sent_info = self._db_select(doc_sent_query)
+            doc_sent_query = "SELECT text FROM sentence WHERE document_id = {0}".format(str(doc_id))
+            doc_sent_text = self._db_select(doc_sent_query)
+
+            if doc_sent_info and doc_sent_text:
+                doc_text = ""
+                for i in range(len(doc_sent_info)):
+                    if i < len(doc_sent_info) - 1:
+
+                        doc_text = doc_text + doc_sent_text[i][0]
+                        doc_sents.append(doc_sent_text[i][0])
+
+                        id_info = doc_sent_info[i][0].split(":")
+                        doc_sents_start_idx.append(id_info[3])
+                        end_index = int(id_info[4])
+
+                        id_info = doc_sent_info[i+1][0].split(":")
+                        start_index = int(id_info[3])
+
+                        diff = start_index - end_index
+                        if diff > 0:
+                            for j in range(diff):
+                                doc_text = doc_text + " "
+
+                doc_text = doc_text + doc_sent_text[len(doc_sent_text)-1][0]
+
+                # adding information of the last sentence
+                id_info = doc_sent_info[len(doc_sent_info) - 1][0].split(":")
+                doc_sents.append(doc_sent_text[len(doc_sent_text)-1][0])
+                doc_sents_start_idx.append(int(id_info[3]))
+
+                return doc_text, doc_sents, doc_sents_start_idx
+            return "", [], []
+        return "", [], []
```

Comment:
If I remove it and in case user passes an arbitrary value as `doc_name`, calling the method will result in the following error: `TypeError: 'NoneType' object is not iterable` please correct me if I'm wrong